types/matrix-js-sdk/index.d.ts

```diff
@@ -337,6 +337,23 @@ export interface SearchBody {
         }
     };
 }
+
+export type SyncState = "PREPARED" | "SYNCING" | "ERROR" | "STOPPED" | "CATCHUP";
+export interface SyncData {
+  /**
+   * The 'next_batch' result from /sync, which will become the 'since' token for the next call to /sync. Only present if state=PREPARED or state=SYNCING.
+   */
+  nextSyncToken?: string;
+  /**
+   * The 'since' token passed to /sync. null for the first successful sync since this client was started. Only present if state=PREPARED or state=SYNCING.
+   */
+  oldSyncToken?: string;
```

Comment:
According to the comment would `null` be a valid value?
```suggestion
  oldSyncToken?: string | null;
```

Reply:
good catch, looked through the sdk's source code, it can be null!